src/Build/Evaluation/Expander.cs

```diff
@@ -1478,12 +1478,25 @@ internal static string ConvertToString(object valueToConvert)
                 else
                 {
                     // The fall back is always to just convert to a string directly.
-                    convertedString = valueToConvert.ToString();
+                    // Issue: https://github.com/dotnet/msbuild/issues/9757
+                    if (IsNumberType(valueToConvert) && ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_10))
+                    {
+                        convertedString = Convert.ToString(valueToConvert, CultureInfo.InvariantCulture.NumberFormat);
```

Comment:
+1 on this if it would achieve goal and wouldn't have obvious downsides. As it would allow us to get rid of the probing for number. And if in future we have other cases (booleans?, time?, etc.), it would be already handled.


> Are we already parsing string->numbers with InvariantCulture?

We should after https://github.com/dotnet/msbuild/pull/8710
So using invariant in opposite dirrection is wanted.

Reply:
I was thinking to cover different types by change wave iterations, to lower the risk of breaking change, however having that we have this change under the change wave, I believe it is safe to enable it for any other data type as well.
And for trying to parse the numeric: 
https://github.com/dotnet/msbuild/blob/00833d9f8772bc99c48ccbebc013aa0a6d5ee622/src/Shared/ConversionUtilities.cs#L187
We are already doing that.

Reply:
Changed the behavior to Convert.ToString(valueToConvert, CultureInfo.InvariantCulture), having that it is under the change wave. two votes for that :)

Reply:
I like this but I think we should hold the bigger change for an early preview of the next release. I am a nervous Nellie!

Reply:
Thank you @rainersigwald. Rolled back the changes to cover only double, long and int.

Reply:
Summary of discussion offline: 
Merge the changes for all types, for early preview of next release.